### PR TITLE
Add Cloud Functions template render sample.

### PR DESCRIPTION
**Pull request by jmdobry**

### CODE REVIEW

**Comment by jmdobry** [comment]
@ace-n Looks good to you?

File: functions/helloworld/index.js [hunk 1]
```diff
@@ -148,3 +148,15 @@ exports.helloError3 = function helloError3 (event, callback) {
 };
 // [END functions_helloworld_error_3]
 /* eslint-enable */
+
+// [START functions_helloworld_template]
+const pug = require('pug');
+
+// Renders the index.pug
```
Comment (ace-n) [inline functions/helloworld/index.js]: Nit: redundant comment

File: functions/helloworld/package.json [hunk 2]
```diff
@@ -22,10 +22,11 @@
   },
   "dependencies": {
     "@google-cloud/debug-agent": "1.0.0",
+    "pug": "2.0.0-beta.12",
     "safe-buffer": "5.0.1"
   },
   "devDependencies": {
-    "@google-cloud/nodejs-repo-tools": "1.3.1",
+    "@google-cloud/nodejs-repo-tools": "1.3.2",
```
Comment (ace-n) [inline functions/helloworld/package.json]: FYI: you'll need to revert this change. (It will cause a merge conflict with #370.)
